apps/server/src/modules/tool/service/validation/school-external-tool-validation.service.ts

```diff
@@ -28,6 +28,22 @@ export class SchoolExternalToolValidationService {
 		this.checkCustomParameterEntries(loadedExternalTool, schoolExternalToolDO);
 	}
 
+	async validateUpdate(schoolExternalToolId: string, schoolExternalToolDO: SchoolExternalToolDO): Promise<void> {
+		this.checkIdMatch(schoolExternalToolId, schoolExternalToolDO.id);
+		this.checkForDuplicateParameters(schoolExternalToolDO);
+		const loadedExternalTool: ExternalToolDO = await this.externalToolService.findExternalToolById(
+			schoolExternalToolDO.toolId
+		);
+		this.checkVersionMatch(schoolExternalToolDO.toolVersion, loadedExternalTool.version);
+		this.checkCustomParameterEntries(loadedExternalTool, schoolExternalToolDO);
+	}
+
+	private checkIdMatch(schoolExternalToolId: EntityId, schoolExternalToolDOId: string | undefined): void {
+		if (schoolExternalToolId !== schoolExternalToolDOId) {
```

Comment:
schoolExternalToolDOId comes from the request body if I see it correctly. There it is marked as optional. Why is it optional if you throw here when it is undefined? Don't you wanna set it required then in the DTO already?

Reply:
We compare the id from the request body(schoolExternalToolDOId ) and the schoolExternalToolId in the path parameter. The variable is optional because it is only used to compare the both id's. The SchoolExternalToolPostParams are also used in the create, where we don't need an id

Reply:
Why do you compare them at all and not rely on one ID, either from the body or the path params?